packages/preset-icons/src/utils.ts

```diff
@@ -1,13 +1,19 @@
 export const isNode = typeof process < 'u' && typeof process.stdout < 'u'
 
+const svgCharRe: [RegExp, string][] = [
+    [/"/g, '\''],
+    [/%/g, '%25'],
+    [/#/g, '%23'],
+    [/{/g, '%7B'],
+    [/}/g, '%7D'],
+    [/</g, '%3C'],
+    [/>/g, '%3E'],
+]
+
 // https://bl.ocks.org/jennyknuth/222825e315d45a738ed9d6e04c7a88d0
 export function encodeSvg(svg: string) {
-  return svg.replace('<svg', (~svg.indexOf('xmlns') ? '<svg' : '<svg xmlns="http://www.w3.org/2000/svg"'))
-    .replace(/"/g, '\'')
-    .replace(/%/g, '%25')
-    .replace(/#/g, '%23')
-    .replace(/{/g, '%7B')
-    .replace(/}/g, '%7D')
-    .replace(/</g, '%3C')
-    .replace(/>/g, '%3E')
+  return svgCharRe.reduce(
+    (str, [regexp, replacement]) => str.replace(regexp, replacement),
+    svg.replace('<svg', (~svg.indexOf('xmlns') ? '<svg' : '<svg xmlns="http://www.w3.org/2000/svg"'))
+  )
```

Comment:
Is this way faster? If not, I would prefer the old one.

Reply:
avoid the replacement when the svg xml namespace present?

Reply:
Reverted since I couldn't find noticeable speed difference